lib/media.js

```diff
@@ -113,15 +168,16 @@ function append(data) {
       throw new Error('Missing html parameter');
     }
 
-    if (mediaMap.styles && mediaMap.styles.length > 0) {
-      html = appendMediaToTop(mediaMap, html);
-    }
-
-    if (mediaMap.scripts && mediaMap.scripts.length > 0) {
-      html = appendMediaToBottom(mediaMap, html);
-    }
+    // Get the combined styles and scripts
+    stylesPromise = combineStyles(mediaMap.styles),
+    scriptsPromise = combineScripts(mediaMap.scripts);
 
-    return html;
+    return Promise.all([stylesPromise, scriptsPromise])
```

Comment:
Because I went the route of ES6 Promises I was thinking it would be better to stick with that implementation.